### ISSUE TITLE
TIME RANGE on TIME column is useless

### ISSUE THREAD

**Issue by frafra** [apache/superset#16870]
Apache Superset should not list `TIME` columns as `TIME COLUMNS` (that is how Superset describes `DATE`/`DATETIME`/`TIME` columns in the `Table` view), or a major rework to handle `TIME` columns properly should be done. Thus, a different issue/PR should be created for that. Setting a time range on such columns is useless at the moment.

#### How to reproduce the bug

1. Go to a dataset containing a `TIME` column
2. Select a time range
3. Run

### Expected results

Such time column should not have been an option, or time range should be able to work with time without a date.

### Actual results

No results.

### Environment

Latest Superset revision on Docker.

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by junlincc** [comment]
My understanding of your request is that : Superset should remove time column selection and allow user to filter data by time range regardless a data has time/date time columns or not. (?)

**Comment by frafra** [comment]
No, I am sorry I have not been able to explain the bug I am experiencing. Nomenclature is confusing.

Superset Table view does not support time-only columns, because it works on datetime or date columns. It makes no sense to a time range selection between dates if you just have hours and minutes for examples. Superset returns an empty result set if such time range filter is applied.

I propose to avoid suggesting time-only columns as possible candidates for the so-called "TIME COLUMN". Otherwise, the time range filter should be able to correctly handle time-only columns, by allowing the user to select a time range filter like "from 00:00 to 08:00" and similar, instead of "between the 26th of January and the 3rd of June" or "last week", which cannot be applied to time-only columns.

**Comment by villebro** [comment]
I agree @frafra , and this is something I've also noticed when working on these parts of the codebase. We should probably make `TIME` "non-temporal" for now, as it really don't support it right now in the time picker. But long term it would be nice to be able to introduce similar functionality for `TIME` columns, e.g. being able to set ranges, just like we do for timestamps.
